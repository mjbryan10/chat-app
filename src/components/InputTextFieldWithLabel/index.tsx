import React from 'react';
import InputTextField from '../InputTextField';
import {Props as InputProps} from '../InputTextField'

interface Props extends InputProps{
  labelValue: string;
}

const InputTextFieldWithLabel: React.FC<Props> = ({labelValue, ...props}) => {
  return (
    <label>
      {labelValue}
      <InputTextField {...props} />
    </label>
  )
}

export default InputTextFieldWithLabel;