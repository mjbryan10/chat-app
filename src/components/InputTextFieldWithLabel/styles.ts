import styled, { ThemeProps, Theme } from 'styled-components';

export const Label = styled.label`
   display: flex;
   flex-flow: column nowrap;
   justify-content: flex-start;
   color: ${({ theme: { color } }: ThemeProps<Theme>) => color.text ?? 'inherit'};
   text-transform: uppercase;
`;
