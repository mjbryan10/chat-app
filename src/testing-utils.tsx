import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }