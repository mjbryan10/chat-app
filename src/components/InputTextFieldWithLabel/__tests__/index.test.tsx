import React from 'react';
import InputTextFieldWithLabel from '..';
import { render, fireEvent, RenderResult, cleanup } from 'testing-utils';

describe('InputTextFieldWithLabel', () => {
   const __testHandler = jest.fn();
   const __labelValue = 'Test';
   let utils: RenderResult;
   beforeEach(() => {
      utils = render(
         <InputTextFieldWithLabel
            handleChange={__testHandler}
            labelValue={__labelValue}
         />
      );
   });
   afterEach(() => cleanup());

   test('should pass down event handler to input element', () => {
      const { getByTestId } = utils;
      const input = getByTestId('input-text-field') as HTMLInputElement;

      fireEvent.change(input, { target: { value: 'Changed text' } });

      expect(input.value).toBe('Changed text');
      expect(__testHandler).toBeCalled();
      expect(__testHandler).toBeCalledWith('Changed text');
   });

   test('should display props labelValue', () => {
      const { getByText } = utils;

      expect(getByText(__labelValue)).toBeVisible();
   });

   test('should render correctly', () => {
      const { asFragment } = render(
         <InputTextFieldWithLabel
            handleChange={__testHandler}
            labelValue={__labelValue}
         />
      );

      expect(asFragment()).toMatchSnapshot();
   });
});
