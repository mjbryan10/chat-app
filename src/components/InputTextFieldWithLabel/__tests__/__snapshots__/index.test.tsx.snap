// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextFieldWithLabel should render correctly 1`] = `
<DocumentFragment>
  <label
    class="sc-AxirZ jSgOQo"
  >
    Test
    <input
      class="sc-AxjAm kuGIAZ"
      data-testid="input-text-field"
      placeholder=""
      value=""
    />
  </label>
</DocumentFragment>
`;
